#! /bin/sh -e
## 50_xemacs_send_buffer_contents by Jens Peter Secher <jps@debian.org>

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0
--- sml-mode-3.9.5.orig/sml-compat.el       31 Jul 2002 23:21:29 -0000      1.2
+++ sml-mode-3.9.5.orig/sml-compat.el       28 Oct 2002 22:00:00 -0000      1.3
@@ -55,7 +55,11 @@
 			(expand-file-name prefix temporary-file-directory)))
 		 (if dir-flag
 		     (make-directory file)
-		   (write-region "" nil file nil 'silent nil 'excl))
+		   (if (featurep 'xemacs)
+                       ;; The 7th arg to `write-region' specifies the
+                       ;; coding system to use in XEmacs.
+                       (write-region "" nil file nil 'silent nil)
+                     (write-region "" nil file nil 'silent nil 'excl)))
 		 nil)
 	    (file-already-exists t))
       ;; the file was somehow created by someone else between
